### PR TITLE
Fix Typescript errors in components/subscriptionCheckouts/stripeForm

### PR DESCRIPTION
**Pull request by i-hardy**
## What are you doing in this PR?

This fixes Typescript errors for the stripeForm component used in the subscription checkouts, and associated files.

It also updates our usage of [the deprecated handleCardSetup](https://stripe.com/docs/js/deprecated/handle_card_setup) to the replacement method [confirmCardSetup](https://stripe.com/docs/js/setup_intents/confirm_card_setup).

[**Trello Card**](https://trello.com/c/llqigKDd)

### CODE REVIEW

File: support-frontend/assets/__mocks__/stripeMock.tsx [hunk 1]
```diff
@@ -1,3 +1,7 @@
+/* eslint-disable eslint-comments/require-description -- This is a mocks file, it is not intended to be good code! */
```
Comment (TomPretty) [inline support-frontend/assets/__mocks__/stripeMock.tsx:1]: 👍

File: support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.tsx [hunk 2]
```diff
@@ -145,19 +143,25 @@ const StripeForm = (props: StripeFormPropTypes) => {
...
-		return stripe.handleCardSetup(clientSecret, cardElement).then((result) => {
-			if (result.error) {
-				handleStripeError(result.error);
-				return Promise.reject(result.error);
-			}
+	const handleCardSetup = (clientSecret: Option<string>) => {
+		const cardElement = elements?.getElement(CardNumberElement);
+		if (stripe && clientSecret && cardElement) {
+			return stripe
+				.confirmCardSetup(clientSecret, {
```
Comment (TomPretty) [inline support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.tsx:150]: here `stripe.handleCardSetup` has changed to `stripe.confirmCardSetup` - is that expected?

Comment (i-hardy) [reply support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.tsx:150]: Yes, as I explained in the intro comment for the PR, `handleCardSetup` is deprecated (and does not have any types), and `confirmCardSetup` is the replacement method. There's links to the Stripe docs on both above.

Comment (TomPretty) [reply support-frontend/assets/components/subscriptionCheckouts/stripeForm/stripeForm.tsx:150]: Oh haha I don't know how I missed that! Nice one 👍

**Review by TomPretty — APPROVED** [review]
Looks great - so many errors squashed! 🔥🔥🔥

**Pull request merged**